localnet/poktrolld/config/supplier1_stake_config.yaml

```diff
@@ -4,7 +4,7 @@ stake_amount: 1000upokt
 services:
   - service_id: anvil
     endpoints:
-      - url: http://anvil.supplier1:8545
+      - url: http://relayminer1:8545
```

Comment:
Why is the `relayminer` endpoint the one backing the `anvil` service?

I thought the `url` for `anvil` should be the "data node" or "service node". No?

Reply:
@Olshansk this is a stake config. That value should be pointing to the endpoint by which relay miner should be accessed to provide service. This staking config was actually never used because `supplier1` is staked in `genesis.json` from the first block (specified in `config.yml`). I just wanted this value to be the same as in `config.yml` to avoid confusion.

Reply:
> which relay miner should be accessed to provide service

Can we reword this sentence to `which the supplier should be accessed to provide service`?

I understand that this works, but I'm already getting confused by why an on-chain advertised address (i.e. for supplying a service) uses a relayminer endpoint (an off-chain endpoint for handling business logic).

For example, `anvil.supplier:8454` was very clear. Seeing how these configs will be used as examples we need to make them clear.

![Screenshot 2024-03-07 at 3 55 17 PM](https://github.com/pokt-network/poktroll/assets/1892194/0b888146-181b-46e5-a367-bd406baa6a95)

Reply:
@Olshansk, this endpoint is a DNS name that was created by Kubernetes/Helm chart, The name comes from the name of the Helm chart deployment [here](https://github.com/pokt-network/poktroll/blob/localnet-scale-up-actors/Tiltfile#L119). This DNS allows all resources within the same Kubernetes namespace (including AppGateServers) to access that RelayMiner.

> why an on-chain advertised address (i.e. for supplying a service) uses a relayminer endpoint (an off-chain endpoint for handling business logic).

I believe this is the purpose of on-chain address advertising: to point to suppliers (RelayMiner services).

In a real-world scenario, when resources are not just deployed in one single Kubernetes namespace, it would be a normal URL such as `supplier-1.poktroll.com`, or `relayminer-1.poktroll.com` or any other DNS.

What am I missing? I can hop on a call to discuss this.

Reply:
Makes sense. In that case.

1. Can we add inline comments in the `xxx_stake_config.yaml` next/above relay miner explaining `the relayminer port is exposed since it proxies the request to the backing anvil data node on the bakend`
2. Can you update https://dev.poktroll.com/configs/supplier_staking_config to explain that it'll usually be the endpoint exposed by the relay miner?

For 2, try to keep it short (not long sentences) but clear w/ bullet points if necessary.